### PR TITLE
feat(dropdown): add keyboard navigation support

### PR DESCRIPTION
**Pull request by wolfgangteves**
This contribution adds keyboard navigation for dropdown component.
Added ArrowUp, ArrowDown navigation, navigates through item list, opens list if necessary.

Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

File: src/dropdown/dropdown.ts [hunk 1]
```diff
@@ -12,11 +17,13 @@ import {NgbDropdownConfig} from './dropdown-config';
     '[class.dropup]': 'up',
     '[class.show]': 'isOpen()',
     '(keyup.esc)': 'closeFromOutsideEsc()',
-    '(document:click)': 'closeFromOutsideClick($event)'
+    '(document:click)': 'closeFromOutsideClick($event)',
+    '(keydown)': 'keyboardEvent($event)'
```
Comment (wesleycho) [inline src/dropdown/dropdown.ts]: I think one thing that may help is binding to the particular keys here, i.e.
```ts
'(keydown.ArrowUp)': 'moveUp()',
'(keydown.ArrowDown)': 'moveDown()'
```
Under the hood, I believe Angular creates only one event listener and does the appropriate things. This avoids the need to pass in the event object and have a bunch of conditional logic in your method.

**Comment by tonyhayes** [comment]
Any idea on when this will be merged into master ?

**Comment by AntonLebiodkin** [comment]
Hi, when keyboard-navigation at dropdown could be added to ng-bootstrap?

**Comment by djohns2021** [comment]
@wolfgangteves @wesleycho any eta on this? The typeahead has this feature while the dropdown lacks it.

**Comment by arajay** [comment]
Still not merged eh?

**Comment by MariMax** [comment]
can someone click merge button, please?

**Comment by aakash1401** [comment]
Any updated on this?

**Comment by maxjoehnk** [comment]
I've reimplemented the proposed changes on the latest master branch (#2683).
I think this PR can be closed as it has huge merge conflicts.

**Comment by pkozlowski-opensource** [comment]
Closing in favour of #2683

**Pull request closed without merging**